flixel/addons/editors/ogmo/FlxOgmo3Loader.hx

```diff
@@ -0,0 +1,514 @@
+package flixel.addons.editors.ogmo;
+
+import flixel.math.FlxPoint;
+import flixel.group.FlxGroup;
+import flixel.FlxG;
+import flixel.tile.FlxTilemap;
+import flixel.math.FlxRect;
+
+using flixel.addons.editors.ogmo.FlxOgmo3Loader;
+using openfl.Assets;
+using haxe.Json;
+using Math;
+using Reflect;
+
+class FlxOgmo3Loader
+{
+	var project:ProjectData;
+	var level:LevelData;
+
+	/**
+	 * Creates a new instance of `FlxOgmoLoader` and prepares the XML level data to be loaded.
+	 * This object can either be contained or ovewritten.
+	 *
+	 * IMPORTANT:
+	 *  * Tile layers must have the Export Mode set to `"CSV"`.
+	 *  * First tile in spritesheet must be blank or debug. It will never get drawn so don't place them in Ogmo!
+	 *    (This is needed to support many other editors that use index `0` as empty)
+	 *
+	 * @param	LevelData	A String or Class representing the location of xml level data.
+	 */
+	public function new(ProjectData:String, LevelData:String)
+	{
+		project = ProjectData.getText().parseProjectJSON();
+		level = LevelData.getText().parseLevelJSON();
+	}
+
+	/**
+	 * Get a custom value for the loaded level.
+	 * Returns `null` if no value is present.
+	 * @param Value 
+	 * @return Dynamic
+	 */
+	public function getLevelValue(Value:String):Dynamic
+	{
+		return level.field(Value);
+	}
+
+	/**
+	 * Load a Tilemap. Tile layers must have the Export Mode set to `"CSV"`.
+	 * Collision with entities should be handled with the reference returned from this function. Here's a tip:
+	 *
+	 * IMPORTANT: Tile layers must export using IDs, not Coords!
+	 *
+	 * @param	TileGraphic		A String or Class representing the location of the image asset for the tilemap.
+	 * @param	TileWidth		The width of each individual tile.
+	 * @param	TileHeight		The height of each individual tile.
+	 * @param	TileLayer		The name of the layer the tilemap data is stored in Ogmo editor, usually `"tiles"` or `"stage"`.
+	 * @return	A FlxTilemap, where you can collide your entities against.
+	 */
+	public function loadTilemap(TileGraphic:Dynamic, TileLayer:String = "tiles"):FlxTilemap
+	{
+		var tilemap = new FlxTilemap();
+		var layer = level.getTileLayer(TileLayer);
+		var tileset = project.getTilesetData(layer.tileset);
+		switch (layer.arrayMode)
+		{
+			case 0:
+				tilemap.loadMapFromArray(layer.data, layer.gridCellsX, layer.gridCellsY, TileGraphic, tileset.tileWidth, tileset.tileHeight);
+			case 1:
+				tilemap.loadMapFrom2DArray(layer.data2D, TileGraphic, tileset.tileWidth, tileset.tileHeight);
+		}
+		return tilemap;
+	}
+
+	/**
+	 * Loads a Map of FlxPoint arrays from a grid layer. For example:
+	 *
+	 * ```haxe
+	 * 	var gridData = myOgmoData.loadGridMap('my grid layer');
```

Comment:
Code blocks shouldn't be indented on top of already being in a fenced code block, otherwise that extra indentation shows up in the API docs.